packages/rpc-methods/src/permitted/middleware.ts

```diff
@@ -0,0 +1,53 @@
+import { logError } from '@metamask/snaps-utils';
+import { ethErrors } from 'eth-rpc-errors';
+import { JsonRpcMiddleware, JsonRpcRequest } from 'json-rpc-engine';
+
+import { handlers } from '.';
+import { selectHooks } from '../utils';
+
+const snapHandlerMap = handlers.reduce((map, handler) => {
+  for (const methodName of handler.methodNames) {
+    map.set(methodName, handler);
+  }
+  return map;
+}, new Map());
+
+/**
+ * Creates a middleware that handles permitted snap RPC methods.
+ *
+ * @param isSnap - A flag that should indicate whether the requesting origin is a snap or not.
+ * @param hooks - An object containing the hooks made available to the permitted RPC methods.
+ * @returns The middleware.
+ */
+export function createSnapMethodMiddleware(
```

Comment:
Maybe this makes more sense?
```suggestion
export function createSnapsMethodMiddleware(
```